### DISCUSSION TITLE
Fix: priority of non-geoblocked channel over geo-blocked ones

### DISCUSSION THREAD

**Discussion by Owyn** [iptv-org/community#146] [Ideas]
### Describe your issue

Currently it only considers a channel geoblocked if it returns "blocked" server answer, but if it returns a custom video \ picture saying "Geo-blocked, please try from another country" - it isn't considered geo-blocked

https://github.com/iptv-org/iptv/commit/a4aa6e7d1e0031f4e4163406d61d61e94436b3be

along with blocked server answer `[geo-blocked]` in title should also mark a channel into geo-blocked category for rivaly purposes 

example:
two Prima channels, where geo-blocked one gets to the public list currently:
https://github.com/iptv-org/iptv/pull/6809/files

**Comment by freearhey** [comment]
this is the expected behavior. in the current version the bot does not use the stream title at all

**Reply by Owyn** [reply]
yea, but it should, that's the problem

you count channels as geo-blocked currently only if they give 403 error which isn't always the case, so using the title is also required for things to work correctly

**Reply by freearhey** [reply]
I understand and that's why I put this post in the "Ideas" category

**Reply by Owyn** [reply]
but it's a bugfix and not an idea thought?
(some geo-blocked channels currently being shown as "online" and not as "geo-blocked" - that's why sorting doesn't work for them https://github.com/iptv-org/iptv/commit/a4aa6e7d1e0031f4e4163406d61d61e94436b3be )

**Comment by what-a** [comment]
@Owyn Currently, the bot _lacks_ programming to differentiate between video of a station and video of a "geo-blocked" announcement. Therefore, we can't fix a bug in the programming (because it does not exist). Instead, we need to brainstorm about new programming ideas to accomplish the task.

**Reply by Owyn** [reply]
>  bot lacks programming to differentiate between video of a station and video of a "geo-blocked" announcement.

I have noticed that those announcements have a pretty peculiar video resolution - 1980x1920
[iptv-checker](https://github.com/freearhey/iptv-checker) (one this project is currently using) - can get stream metadata including height & width,
what do you think about marking a stream as "blocked" instead of online when that video resolution is found?
